torch/distributed/distributed_c10d.py

```diff
@@ -1400,7 +1412,7 @@ def all_gather_object(object_list, obj, group=group.WORLD):
     input_tensor.resize_(max_object_size)
     coalesced_output_tensor = torch.empty(
         max_object_size * group_size, dtype=torch.uint8
-    ).to(my_rank if is_nccl_backend else "cpu")
+    ).to(torch.cuda.current_device() if is_nccl_backend else "cpu")
```

Comment:
nit: use a var to dedup this if-else and also merge it with the `if is_nccl_backend` clause? E.g.:

```python
current_device = torch.device("cpu")
if is_nccl_backend:
    current_device = torch.cuda.current_device()
    input_tensor = input_tensor.to(current_device)
    ...
object_sizes_tensor = torch.zeros(group_size, dtype=int, device=current_device)
...
```